networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -301,4 +289,19 @@ public BeaconChainMethods getBeaconChainMethods() {
   private boolean peerIsReady(Eth2Peer peer) {
     return peer.hasStatus();
   }
+
+  public SafeFuture<?> sendGoodbyeToPeers() {
+    return SafeFuture.allOf(
+        connectedPeerMap.values().stream()
+            .map(
+                peer ->
+                    peer.sendGoodbye(GoodbyeMessage.REASON_CLIENT_SHUT_DOWN)
+                        .thenRun(() -> peer.disconnectCleanly(DisconnectReason.SHUTTING_DOWN))
```

Comment:
You don't need to sendGoodbye then disconnectCleanly - disconnectCleanly sends the goodbye.  It probably does need to be adjusted to return a future so you know when it's actually been written.

Reply:
Done.